contracts/cosmwasm-vm/cw-integration/tests/test_host_lightclient.rs

```diff
@@ -206,13 +216,32 @@ pub fn call_receive_packet(ctx: &mut TestContext) -> Result<AppResponse, AppErro
     )
 }
 
-fn call_bind_port(ctx: &mut TestContext, port_name: &str) -> Result<AppResponse, AppError> {
+fn call_bind_port(
+    ctx: &mut TestContext,
+    port_name: &str,
+    addr: &str,
+) -> Result<AppResponse, AppError> {
     ctx.app.execute_contract(
         ctx.sender.clone(),
         ctx.get_ibc_core(),
         &CoreMsg::ExecuteMsg::BindPort {
             port_id: port_name.to_string(),
-            address: ctx.get_xcall_app().to_string(),
+            address: addr.to_string(),
+        },
+        &[],
+    )
+}
+
+fn call_xcall_app_message(ctx: &mut TestContext, data: Vec<u8>) -> Result<AppResponse, AppError> {
+    ctx.app.execute_contract(
+        Addr::unchecked("archway1q6lr3hy5cxk4g74k9wcqyqarf9e97ckpn7t963"),
```

Comment:
Why is this address hardcoded ? Is it specific to any environment or doesn't matter ?

Reply:
ya because the address is also rlp encoded in the payload data of xcall so we had to hardcode this for this specific data.